### DISCUSSION TITLE
TornadoMath.atan2 plugin has incorrect type

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [beehive-lab/TornadoVM#480] [Q&A]
Hi.
I noticed that TornadoMath.atan2 plugin is registered using double,double type parameters but in reality it uses float,float ones.
So my question is, does it work in the sense is this code executed on GPU?

**Comment by andrii0lomakin** [comment]
Ah, I found that plugins are registered both for double,double and float,float methods. 
So, the second one is merely useless.

**Reply by andrii0lomakin** [reply]
If I got it correctly. And not second but first one of course.

**Comment by jjfumero** [comment]
Good catch. The `atan` function is registered for both `float` and `double` from the `TornadoMath` class. However we missed the `double`, `double` signature. We will provide those. Thanks!

**Reply by andrii0lomakin** [reply]
Do you need to have  a double version ? I thought that GPUs do not widely support doubles. That will create  false impression about the precision of the calculation for user. Do not you think so ?

**Reply by jjfumero** [reply]
Yes, most modern NVIDIA discrete GPUs support `FP64`, and we have some TornadoVM users doing HPC and data analytics that require `FP64`.  For the precision, that's why we narrow to `FP32` via the `TornadoMath` class instead of the `Math` class. But we are open to suggestions.

**Reply by andrii0lomakin** [reply]
I mean, in your response, you wrote that you will provide a double version. IMHO better to do not provide double,double version but remove plugin for double signature. In general, I suggest removing versions for methods that manipulate doubles or at least deprecate them because otherwise, it creates a false impression about the precision of the level of calculation.

**Reply by jjfumero** [reply]
If you call the `TornadoMath` with the double version, the TornadoVM JIT compiler generates `fp64` operations. So, it is supported. As a developer, you can choose which one to invoke, the `float` or `double`. If you do not see the corresponding pairs `double` and `float` in the `TornadoMath` class, it is probably a miss from our side.  TornadoVM never narrows down automatically, at least at the moment`.

**Reply by andrii0lomakin** [reply]
I see, got it. Thank you.
